test/toolkit.js

```diff
@@ -124,3 +124,81 @@ test('h.file() outside default confine', async () => {
         status  : 403
     }));
 });
+
+test('h.directory() serve files', async () => {
+    const server = pogo.server();
+    server.route({
+        method : 'GET',
+        path   : '/dirtest/{file*}',
+        handler(request, h) {
+            return h.directory('./test/fixture');
+        }
+    });
+    const response = await server.inject({
+        method : 'GET',
+        url    : '/dirtest/names.json'
+    });
+    assertStrictEq(response.status, 200);
+    assertStrictEq(response.headers.get('content-type'), 'application/json; charset=utf-8');
+    assertEquals(response.body, new TextEncoder().encode('[\n    "Alice",\n    "Bob",\n    "Cara"\n]\n'));
+});
+
+test('h.directory() listing with relative path', async () => {
+    const server = pogo.server();
+    server.route({
+        method : 'GET',
+        path   : '/dirtest/{dirname?}',
+        handler(request, h) {
+            return h.directory('./test/fixture', {listing: true});
+        }
+    });
+    const response = await server.inject({
+        method : 'GET',
+        url    : '/dirtest/'
+    });
+    assertStrictEq(response.status, 200);
+    assertStrictEq(response.headers.get('content-type'), 'text/html; charset=utf-8');
+    assertStringContains(response.body, '<html>');
+    assertStringContains(response.body, 'names.json');
+});
+
+test('h.directory() listing with partial relative path', async () => {
+    const server = pogo.server();
+    server.route({
+        method : 'GET',
+        path   : '/dirtest/{dirname?}',
+        handler(request, h) {
+            return h.directory('test/fixture', {listing: true});
+        }
+    });
+    const response = await server.inject({
+        method : 'GET',
+        url    : '/dirtest/'
+    });
+    assertStrictEq(response.status, 200);
+    assertStrictEq(response.headers.get('content-type'), 'text/html; charset=utf-8');
+    assertStringContains(response.body, '<html>');
+    assertStringContains(response.body, 'names.json');
+});
+
+test('h.directory() forbids listing by default', async () => {
```

Comment:
Can you move this test so it is before the `listing: true` tests? (Better to have default cases before edge cases.)

Reply:
sure